core/src/subgraph/instance.rs

```diff
@@ -24,22 +21,41 @@ impl<T> SubgraphInstanceTrait<T> for SubgraphInstance<T>
 where
     T: RuntimeHostBuilder,
 {
-    fn from_manifest(logger: &Logger, manifest: SubgraphManifest, host_builder: T) -> Self {
+    fn from_manifest(
+        logger: &Logger,
+        manifest: SubgraphManifest,
+        host_builder: T,
+    ) -> Result<Self, Error> {
         // Create a new runtime host for each data source in the subgraph manifest;
         // we use the same order here as in the subgraph manifest to make the
         // event processing behavior predictable
-        let hosts = manifest
+        let manifest_id = manifest.id;
+        let (hosts, errors): (_, Vec<_>) = manifest
             .data_sources
-            .iter()
-            .map(|d| host_builder.build(&logger, manifest.clone(), d.clone()))
-            .map(Arc::new)
-            .collect();
+            .into_iter()
+            .map(|d| host_builder.build(&logger, manifest_id.clone(), d))
+            .partition(|res| res.is_ok());
 
-        // Remember manifest and managed hosts
-        SubgraphInstance {
-            _manifest: manifest,
-            hosts: hosts,
+        if !errors.is_empty() {
+            let joined_errors = errors
+                .into_iter()
+                .map(Result::unwrap_err)
+                .map(|e| e.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+            return Err(format_err!(
+                "errors starting data sources: {}",
```

Comment:
Not with the term by itself but I _think_ (may be wrong) we're not talking about data sources as something that we `start`, `run` or so, unlike with mappings. I'm undecided here though.

Reply:
The only error this can currently return is "No ABI entry found for the main contract of data source", which is about the data source itself, not about the mapping. So if a user is trying to debug this, "data source" is the more helpful term imo.

Reply:
How about `Errors loading data sources` then?

Reply:
I like it.